docgenerator/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+find_package(Doxygen REQUIRED)
+find_package(Sphinx REQUIRED)
+
+# adapted from https://devblogs.microsoft.com/cppblog/clear-functional-c-documentation-with-sphinx-breathe-doxygen-cmake/
+# Find all headers
+file(GLOB_RECURSE HEADERS ${PROJECT_SOURCE_DIR}/*.hxx)
+
+set(DOXYGEN_INPUT_DIR ${PROJECT_SOURCE_DIR}/)
+set(DOXYGEN_OUTPUT_DIR ${PROJECT_SOURCE_DIR}/docs/doxygen/)
```

Comment:
The output dir must be in the build directory. Never touch the source directory!